app/models/tenant.rb

```diff
@@ -317,6 +318,10 @@ def create_miq_product_features_for_tenant_nodes
     MiqProductFeature.seed_single_tenant_miq_product_features(self)
   end
 
+  def destroy_with_subtree
```

Comment:
Could you change the title and possibly the description a little.

The wording suggests there was a business reason for not allowing delete. But the PR suggests that it is broken. I'm a little confused.

I wonder if `:orphan_strategy =>:destroy` would do this for you?

Reply:
we do not want to allow to delete tenant/project who has children over UI, BUT
for cloud provider we are creating tenant's tree automatically based on `CloudTenant` and when "cloud ems" destroyed we want to destroy all tenant created automatically.

This new method will be called only as callback when `cloud_manager` destroyed

Reply:
@kbrock changed PR description

Reply:
Think Keenan's right about using `:orphan_strategy` but instead of `:destroy` maybe consider using `:restrict`: `:restrict  An AncestryException is raised if any children exist`. If we did this, the callback wouldn't be necessary. The error that gets raised is this: `Ancestry::AncestryException (Cannot delete record because it has descendants)` which seems like something we should be using not just on tenants (maybe also the rest of the ancestry-using models.)

Reply:
`has_ancestry(:orphan_strategy => :restrict)` is the model line.

Reply:
callback still needed since it is checking for other conditions than having children

Reply:
My idea would help with the performance of this regardless.

Reply:
👍